test/test_mongo_middleware.js

```diff
@@ -238,6 +240,48 @@ describe('mongo db middleware', function() {
       });
     });
 
+    it('passes isForSubmit = true in options when fields has $submit = true', function(done) {
+      sinon.spy(Collection.prototype, 'find');
+      var middlewareSpy = sinon.spy(function(request, next) {
+        expect(request).to.have.all.keys([
+          'action',
+          'collectionName',
+          'options',
+          'query'
+        ]);
+        expect(request.action).to.equal(BEFORE_SNAPSHOT_LOOKUP);
+        expect(request.collectionName).to.equal('testcollection');
+        expect(request.options.testOptions).to.equal('yes');
+        expect(request.options.isForSubmit).to.equal(true);
+        expect(request.query._id).to.equal('test1');
+        // test that when we set queryOptions in the middleware, they get passed to the Mongo driver.
+        request.queryOptions = {
+          isForSubmit: true
+        };
+        next();
+      });
+      db.use(BEFORE_SNAPSHOT_LOOKUP, middlewareSpy);
+
+      var snapshot = {type: 'json0', id: 'test1', v: 1, data: {foo: 'bar'}};
+      db.commit('testcollection', snapshot.id, {v: 0, create: {}}, snapshot, null, function(err) {
+        if (err) return done(err);
+        db.getSnapshot('testcollection', 'test1', {
+          $submit: true
+        }, {testOptions: 'yes'}, function(err, doc) {
+          if (err) return done(err);
+          expect(doc).to.exist;
+          expect(middlewareSpy.calledOnceWith(sinon.match.object, sinon.match.func)).to.equal(true);
+          expect(Collection.prototype.find.calledOnceWith({
```

Comment:
(If we want more fluent tests, we could add a dependency on [`sinon-chai`](https://www.chaijs.com/plugins/sinon-chai/))